specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Dataset.json

```diff
@@ -166,7 +166,7 @@
     },
     "JsonFormatFilePattern": {
       "description": "JSON format file pattern. A property of JsonFormat.",
-      "type": "string",
```

Comment:
Why all those changes? Those properties are modeled as enums with `modelAsString` set to true.

Reply:
Hi Kpajdzik, after that, I will consult the feature owner, which may take some time.

Reply:
What do you mean by "after that"?

Reply:
This change is to adapt with the private swagger validation on accepting the value as expressions. @hvermis , could you please help share some insights on Kpajzik's comments here? it seems I could not find Sparsh. Thanks.

Reply:
@kpajdzik We have Expressions in ADF with special language that our runtime understands. Unfortunately in Swagger 2.0 we can't model it as oneOf since it doesn't support, so we model the properties that accept expressions as object to allow both types.

Reply:
I think in that case you need to remove enum property, if you want to have those parameters as objects. Enum constraint can be only applied to string type.